detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -122,14 +120,18 @@ class AndroidDriver extends DeviceDriverBase {
     return pid;
   }
 
-  async deliverPayload(params) {
-    const {delayPayload, url} = params;
+  async deliverPayload(params, deviceId) {
+    if (params.delayPayload) {
+      return;
+    }
```

Comment:
`delayPayload` is irrelevant for Android, it seems. We ignore everything when it is set; rather, we look at the explicit params provided to `launchApp()` (where we used to get the previously cached url, previously)